lib/floki.ex

```diff
@@ -304,25 +304,68 @@ defmodule Floki do
     |> Enum.map(fn html_node -> HTMLTree.to_tuple(tree, html_node) end)
   end
 
+  @deprecated """
+  Use `find_and_update/3` or `Enum.map/2` instead.
+  """
+  def map(_html_tree_or_list, _fun)
+
+  def map(html_tree_list, fun) when is_list(html_tree_list) do
+    Enum.map(html_tree_list, &Finder.map(&1, fun))
+  end
+
+  def map(html_tree, fun), do: Finder.map(html_tree, fun)
+
   @doc """
-  It receives a HTML tree structure as tuple and maps
-  through all nodes with a given function that receives
-  a tuple with {name, attributes}.
+  Searchs for elements inside the HTML tree and update those that matches the selector.
 
-  It returns that structure transformed by the function.
+  It will return the updated HTML tree.
 
-  ## Examples
+  This function works in a way similar to `traverse_and_update`, but instead of updating
+  the children nodes, it will only updates the `tag` and `attributes` of the matching nodes.
 
-      iex> html = {"div", [{"class", "foo"}], ["text"]}
-      iex> Floki.map(html, fn({name, attrs}) -> {name, [{"data-name", "bar"} | attrs]} end)
-      {"div", [{"data-name", "bar"}, {"class", "foo"}], ["text"]}
+  If `fun` returns `nil`, the HTML node will be removed from the tree.
 
+  ## Examples
+
+      iex> Floki.find_and_update([{"a", [{"href", "http://elixir-lang.com"}], ["Elixir"]}], "a", fn
+      iex>   {"a", [{"href", href}]} ->
+      iex>     {"a", [{"href", String.replace(href, "http://", "https://")}]}
+      iex>   other ->
+      iex>     other
+      iex> end)
+      [{"a", [{"href", "https://elixir-lang.com"}], ["Elixir"]}]
   """
-  def map(html_tree_list, fun) when is_list(html_tree_list) do
-    Enum.map(html_tree_list, &Finder.map(&1, fun))
-  end
 
-  def map(html_tree, fun), do: Finder.map(html_tree, fun)
+  @spec find_and_update(
+          html_tree(),
+          Finder.selector(),
+          ({String.t(), [html_attribute()]} -> {String.t(), [html_attribute()]} | nil)
+        ) :: html_tree()
+  def find_and_update(html_tree, selector, fun) do
+    {tree, results} = Finder.find(html_tree, selector)
+
+    operations_with_nodes =
+      Enum.map(results, fn
+        html_node = %Floki.HTMLTree.HTMLNode{} ->
+          case fun.({html_node.type, html_node.attributes}) do
+            {updated_tag, updated_attrs} ->
+              {:update, %{html_node | type: updated_tag, attributes: updated_attrs}}
+
+            nil ->
+              {:delete, html_node}
+
+            _ ->
```

Comment:
Are you sure you want to ignore invalid results? :D

Reply:
I'm not sure /o\
Maybe letting it crash for invalid match? :thinking:

Reply:
:+1: for let it crash!